### PR TITLE
Fix: Filter providers based on viewport mobile

### PR DESCRIPTION
**Pull request by kamikazebr**
Closes https://github.com/1Hive/gardens/issues/504

It uses viewport to check if it in mobile. The use the viewport to check if on mobile its used across all base code, but in tha case it should use some other method to make that check, allowing for example after resized the window on PC we still be able see all wallet providers.

image

image

## How test?

1. Open the vercel preview link on Desktop with full screen browser
2. Click in Enable Account
3. It should show all providers
4. Resize the window until the button change
5. Click in the Button Enable Account
6. It should filter the providers and show only WalletConnect

### CODE REVIEW

File: src/ethereum-providers/connectors.ts [hunk 1]
```diff
@@ -40,12 +40,21 @@ export const CONNECTORS = [
         },
       }
     : null,
-].filter(p => p)
```
Comment (tmoutinho) [inline src/ethereum-providers/connectors.ts:43]: Same comment as above.

**Review by tmoutinho — COMMENTED** [review]
Left some comments.

File: src/ethereum-providers/connectors.ts [hunk 2]
```diff
@@ -40,12 +40,21 @@ export const CONNECTORS = [
...
+].filter((p) => p)
+
+export const CONNECTORS_MOBILE = CONNECTORS.filter(
```
[the remark this answers is not in the document]
Comment (kamikazebr) [reply src/ethereum-providers/connectors.ts:45]: Excelent idea!

File: src/providers/Wallet.js [hunk 3]
```diff
@@ -18,6 +18,7 @@ import {
 import { getPreferredChain, setPreferredChain } from '@/local-settings'
 import { useWalletConnectors } from '@/ethereum-providers/connectors'
 
+import { useViewport } from '@1hive/1hive-ui'
```
[the remark this answers is not in the document]
Comment (kamikazebr) [reply src/providers/Wallet.js:21]: I tried, that thing got wild! Alot files need be changed too to do it. So i keep it simple.

While was converting thats files to Typescript i got alot weirds stuffs. For example a method connect() need 1 argument but was been called with 0 args and alots another things. 

Maybe we create new PR for it?

File: src/components/Account/ScreenProviders.js [hunk 4]
```diff
@@ -8,18 +8,22 @@ import {
   RADIUS,
   textStyle,
```
[the remark this answers is not in the document]
Comment (kamikazebr) [reply src/components/Account/ScreenProviders.js:9]: Yah that i think its more possible to do, will check

File: src/ethereum-providers/connectors.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kamikazebr) [reply src/ethereum-providers/connectors.ts:45]: BTW I really dont know why it need be there, that filter, maybe to remove the null PORTIS when it was javascript file. Almost sure it not more necessary

File: src/components/Account/ScreenProviders.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (kamikazebr) [reply src/components/Account/ScreenProviders.js:8]: Done

File: src/components/Account/ScreenProviders.tsx [hunk 5]
```diff
@@ -78,11 +87,21 @@ function ScreenProviders({ onActivate }) {
...
   )
 }
-ScreenProviders.propTypes = {
-  onActivate: PropTypes.func.isRequired,
+
+type ProviderButtonType = {
+  id: string
+  provider?: {
+    image: string
+    name: string
```
Comment (kamikazebr) [inline src/components/Account/ScreenProviders.tsx:95]: Its need be optional, required by other lib

File: src/components/Account/ScreenProviders.tsx [hunk 6]
```diff
@@ -110,25 +129,21 @@ function ProviderButton({ id, provider, onActivate }) {
...
+          <div
+            css={`
+              margin-top: ${1 * GU}px;
+              ${textStyle('body1')};
+            `}
+          >
+            {provider.name}
+          </div>
+        </>
+      )}
```
Comment (kamikazebr) [inline src/components/Account/ScreenProviders.tsx:144]: Provider now can be undefined, here i choose just not show anything while provider its not defined.

Using 'if else' with message for example "Provider not found" could lead to flickering the UI.

File: src/ethereum-providers/connectors.ts [hunk 7]
```diff
@@ -6,7 +6,20 @@ const RINKEBY_ETH_NODE = env('RINKEBY_ETH_NODE')
 const XDAI_ETH_NODE = env('XDAI_ETH_NODE')
 const POLYGON_ETH_NODE = env('POLYGON_ETH_NODE')
 
-export const CONNECTORS = [
+type ConnectorProviderType = {
+  [key: string]: any
+  id: string
+  properties: {
+    [key: string]: any
```
Comment (kamikazebr) [inline src/ethereum-providers/connectors.ts:13]: Strong types already knowed but with dynamic to ensure it now will break anything

File: src/ethereum-providers/connectors.ts [hunk 8]
```diff
@@ -31,21 +44,31 @@ export const CONNECTORS = [
...
+]
+
+if (PORTIS_ID) {
+  CONNECTORS.push({
+    id: 'portis',
+    properties: {
+      dAppId: PORTIS_ID,
+      chainId: [100, 4],
+    },
+  })
```
Comment (kamikazebr) [inline src/ethereum-providers/connectors.ts:56]: Extract that IF to correct the return type. it was retuning union with null and getting errors on other file use that.

**Review by tmoutinho — APPROVED** [review]
✌️

**Review by 0xGabi — CHANGES_REQUESTED** [review]
LGTM!

**Review by 0xGabi — APPROVED** [review]

**Pull request merged**